### PR TITLE
Issue #211 - Handle Collections and Maps containing JsonValues

### PR DESCRIPTION
**Pull request by toddobryan**
JsonArrayBuilder and JsonObjectBuilder add methods handle
JsonValues and JsonArrayBuilder and JsonObjectBuilder instances,
but creating builders using a Collection or Map that contains
these values fails, because MapUtil does not handle them and
throws an IllegalArgumentException saying that the types are
not supported.

This commit includes tests that create builders with such
values and changes to MapUtil to handle them. (It also
refactors MapUtil to avoid unnecessary excessive nesting.)

Signed-off-by: Todd O'Bryan <toddobryan@gmail.com>

### CODE REVIEW

**Comment by m0mus** [comment]
@lukasj please review

**Comment by toddobryan** [comment]
Is anyone going to review this? I'm starting to think the reference implementation is kinda dead...

**Comment by keilw** [comment]
@toddobryan Keep in mind, last week was EclipseCon Europe and before that everyone was busy with the Jakarta EE 8 overall release. Whether JSONP shall add new features in Jakarta EE 9 or just do the refactoring from `javax.json` to `jakarta.json` is not yet decided either,

File: impl/src/main/java/org/glassfish/json/MapUtil.java [hunk 1]
```diff
@@ -18,14 +18,15 @@
...
-import javax.json.JsonValue;
 import java.math.BigDecimal;
 import java.math.BigInteger;
 import java.util.Collection;
 import java.util.Map;
 
+import javax.json.JsonArrayBuilder;
+import javax.json.JsonObjectBuilder;
+import javax.json.JsonValue;
+
```
Comment (lukasj) [inline impl/src/main/java/org/glassfish/json/MapUtil.java]: What is the point of this change? Is it required for everyone to update existing imports to fit the setup of his IDE or should the one doing a change update his/her IDE to follow conventions used by the project? I have nothing against changing wildcard imports to fqn ones but I don't like every other change to come up with re-sorted imports

Comment (toddobryan) [reply impl/src/main/java/org/glassfish/json/MapUtil.java]: That was accidental. Sorry about that.

Comment (toddobryan) [reply impl/src/main/java/org/glassfish/json/MapUtil.java]: OK, imports are reverted back to their previous state. Sorry that it took me so long to get to this after complaining about you taking so long, but I actually gave up and switched to a different implementation.

Also, I couldn't find a style guide or coding conventions anywhere. I should have caught the change in imports, but it's hard to set my IDE prefs if I don't know what they should be set to.

**Comment by toddobryan** [comment]
OK. I have no idea what I did. Yesterday, I made changes within Github, because I was at work and only have the project on my personal machine. Turns out I needed to sign those commits, which I gather I can't do just in Github. Today, I tried to grab the changes that were made to master since my fork/pull request was done, but it looks like they're showing up as new changes.

Here's the bug this is trying to fix.
```
ArrayList<JsonValue> list = new ArrayList<>();
list.add(JsonValue.TRUE);
Json.createArrayBuilder(list).build();
```
You get a nasty error because the builder implementations that take a `Collection` or a `Map` know how to handle things like `String` and `Boolean`, but can't handle `JsonString` or `JsonValue.TRUE` or `JsonValue.FALSE`. They also can't handle `JsonArrayBuilder` or `JsonObjectBuilder`.

I've since switched to using the `joy` implementation at https://github.com/leadpony/joy. That implementation also couldn't handle `JsonArrayBuilder` or `JsonObjectBuilder` and I posted some code that didn't compile and was missing imports. The maintainer fixed my code and merged it within 24 hours of my post. As an added bonus, that implementation uses the Apache 2 license. Maybe that implementation should become the reference implementation and the glassfish one should be allowed to die a quiet death?

If not, feel free to use or not use the code I've posted. The important lines are
https://github.com/toddobryan/jsonp/blob/master/impl/src/main/java/org/glassfish/json/MapUtil.java#L43-L48
Everything else is just tests.

If someone can tell me an easy way to save this PR, I'm happy to do it, but I can't care more about fixing bugs than the maintainers.

**Comment by lukasj** [comment]
replaced by #252

**Pull request closed without merging**